### PR TITLE
Sourcify integration

### LINKED ISSUE #209: Integrate Sourcify with Blockscout

### PR DESCRIPTION
**Pull request by aaronmboyd**
## Motivation

Fixes #209 

Cherry-picked from `blockscout-premium/vb-premium-sourcify-integration` (private branch) with many conflicts and bulk changes fixed by hand.

### Enhancements
* Add Sourcify as code verification target with a file picker and dropzone for compiled Solidity JSON files (compiled with metadata)

### Bug Fixes
* There were some files that may simply be a reflection of the code styles that have diverged between blockscout and celo-org/blockscout, in particular some modules not being aliased the same, so `Chain.<something>` became `Explorer.Chain.<something>`.

### Incompatible Changes
* All existing verification views and code remain, feature toggled off.

## Upgrading
* New database migration script in this release. Adds a single boolean column to the contracts table "verified".
* To feature toggle Sourcify verification, set the following environment variables:
```
ENABLE_SOURCIFY_INTEGRATION=true
SOURCIFY_SERVER_URL=https://sourcify.dev/server 
SOURCIFY_REPO_URL=https://repo.sourcify.dev/contracts/full_match/ 
CHAIN_ID=44787 
```

## Checklist for your Pull Request (PR)

Checklist not completed, will test manually in Alfajores.
CHANGELOG.md not being maintained by `celo-org/blockscout`.

  - [ ] I added an entry to `CHANGELOG.md` with this PR
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

File: apps/explorer/lib/explorer/chain.ex [hunk 1]
```diff
@@ -3044,6 +3116,69 @@ defmodule Explorer.Chain do
...
+    case Repo.get(Address, address_hash) do
+      nil ->
+        %{:verified_contract => nil, :additional_sources => nil}
+
+      target_address ->
+        target_address_hash = target_address.hash
+        contract_code = target_address.contract_code
+
+        case contract_code do
+          %Explorer.Chain.Data{bytes: contract_code_bytes} ->
```
[the remark this answers is not in the document]
Comment (aaronmboyd) [reply apps/explorer/lib/explorer/chain.ex:3133]: Thanks, will make the changes

File: docker/Dockerfile [hunk 2]
```diff
@@ -20,8 +20,8 @@ RUN mix do deps.get, deps.compile
 ADD . .
 
 ARG COIN
-ARG DISABLE_WRITE_API="false"
-ARG DISABLE_INDEXER="false"
+ARG DISABLE_WRITE_API="true"
+ARG DISABLE_INDEXER="true"
```
Comment (carterqw2) [inline docker/Dockerfile:24]: Inverting these values will break the build of the indexer and web images in production as it changes the default value, if you need to build a docker image with different arguments, you could pass them using `--build-arg DISABLE_WRITE_API=true`

**Review by carterqw2 — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 2 above)
Comment (aaronmboyd) [reply docker/Dockerfile:24]: Ahh yes! This was a mistake, I was changing things during my local development and left it checked in. I will revert.

File: apps/block_scout_web/assets/css/app.scss [hunk 3]
```diff
@@ -117,6 +117,9 @@ $fa-font-path: "~@fortawesome/fontawesome-free/webfonts";
 @import "components/custom_tooltips_block_details";
 @import "components/_erc721_token_image_container";
 @import "components/_inventory_token_instance_image_container";
+// @import "components/_external_link";
```
[the remark this answers is not in the document]
Comment (aaronmboyd) [reply apps/block_scout_web/assets/css/app.scss:120]: Will do, I left a few comments in place but best to delete.

**Comment by aaronmboyd** [comment]
image
@mrsmkl @carterqw2 Any clues how to fix this?

I can see in `.../blockscout/apps/block_scout_web/lib/block_scout_web/views/address_view.ex`
```
  def smart_contract_verified?(%Address{smart_contract: %SmartContract{}}), do: true

  def smart_contract_verified?(%Address{smart_contract: nil}), do: false
  ```
 So the function should exist?

**Comment by aaronmboyd** [comment]
Some remaining issues related to the Ecto sandbox connections, was reading through this - https://hexdocs.pm/ecto_sql/Ecto.Adapters.SQL.Sandbox.html

Do we need to change the connection pooling mode?

File: apps/explorer/lib/explorer/chain.ex [hunk 4]
```diff
@@ -876,6 +877,37 @@ defmodule Explorer.Chain do
     Wei.to(gas_price, unit)
   end
 
+  defp augment_celo_address(orig_address) do
```
Comment (carterqw2) [inline apps/explorer/lib/explorer/chain.ex:880]: Is it used anywhere?

**Review by carterqw2 — APPROVED** [review]

**Pull request merged**